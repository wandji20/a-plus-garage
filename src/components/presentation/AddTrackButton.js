import React from 'react';
// import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const AddTrackButton = () => {
  // const { style } = props;
  const style = {
    display: 'block',
    marginTop: '200px',
  };
  return (
    <Link to="/new_car" style={style}>
      <button type="button" className="btn btn-info">
        Track Car
      </button>
    </Link>
  );
};

// AddTrackButton.propTypes = {
//   style: PropTypes.objectOf(PropTypes.string),
// };

// AddTrackButton.defaultProps = {
//   style: {},
// };

export default AddTrackButton;
