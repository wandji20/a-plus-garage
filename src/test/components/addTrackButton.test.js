// import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import AddTrackButton from '../../components/presentation/AddTrackButton';
import { BrowserRouter } from 'react-router-dom';

it('renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}>
        <BrowserRouter>
          <AddTrackButton />
        </BrowserRouter>
      </Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
