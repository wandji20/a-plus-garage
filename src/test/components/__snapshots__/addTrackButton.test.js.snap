// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  href="/new_car"
  onClick={[Function]}
  style={
    Object {
      "display": "block",
      "marginTop": "200px",
    }
  }
>
  <button
    className="btn btn-info"
    type="button"
  >
    Track Car
  </button>
</a>
`;
